### DISCUSSION TITLE
Development Release 3.0.0dr1

### DISCUSSION THREAD

**Discussion by campreilly** [campreilly/UnderSeaModelingLibrary#262] [Announcements]
**Major re-design for Development Release 3.0.0dr1 (https://github.com/campreilly/UnderSeaModelingLibrary/pull/261)**

Migrate code to C++17 standard

- Upgrade to gcc 11.3, CMake 3.20, Boost 1.82, necdf 4.8
- Replace Boost shared pointers and thread utilities with C++17 equivalents.
- Use clang-tidy to automatically migrate code to C++17 best practices.
- Use clang-format to reformat code to follow Google coding standards.
- Use Eclipse "organize includes" to make dependencies explicit.

Enhancements

- Replace seq_data with immutable algorithms.
- Make ocean objects immutible and replace memory management scheme with constant shared pointers.
- Rework platforms, sensors, beampatterns, eigenrays, eigenverbs, and biverbs to better fit C++17 best practices.
- Add doxygen build target and ignore BOOST deprecated headers in CMake build scripts.
- Update style documentation in Mainpage.h
- Use git lfs to store large data files on GitHub.

**Comment by cconvey** [comment]
Which version of "clang-format" are you using?

Last time I checked, the behavior of "clang-format" actually changes from one release to the next.

When I run `clang-format-15 --dry-run **/*.cc **/*.h`, I'm seeing a _lot_ of warnings about non-compliant formatting.
